common/file.c

```diff
@@ -374,38 +374,30 @@ ods_replace(const char *str, const char *oldstr, const char *newstr)
         return buffer;
     }
 
+    // Length of str upto the first occurence of oldstr
     part1_len = ch-str;
+
+    // Length of newstr (which will be inserted after part 1)
     part2_len = strlen(newstr);
-    part3_len = strlen(ch+strlen(oldstr));
+
+    // Length of the string that follows oldstr
+    char* part3 = ch+strlen(oldstr);
+    part3_len = strlen(part3);
+
     buffer = calloc(part1_len+part2_len+part3_len+1, sizeof(char));
     if (!buffer) {
         return NULL;
     }
+    buffer[0] = '\0';
 
-    if (part1_len) {
-        strncpy(buffer, str, part1_len);
-        buffer[part1_len] = '\0';
-
-        if (part2_len) {
-            strncat(buffer, str, part2_len);
-            buffer[part1_len+part2_len] = '\0';
-        }
-    } else {
-        strncpy(buffer, newstr, part2_len);
-        buffer[part2_len] = '\0';
-    }
-
-    if (part3_len) {
-        strncat(buffer, ch+strlen(oldstr), part3_len);
-        buffer[part1_len+part2_len+part3_len] = '\0';
-    }
+    // Characters preceeding oldstr exist in str. Copy them to the output buffer.
+    strncat(buffer, str, part1_len);
+    strcat(buffer, newstr);
+    strcat(buffer, part3);
```

Comment:
```suggestion
    strlcat(buffer, str, part1_len+part2_len+part3_len+1);
    strlcat(buffer, newstr, part1_len+part2_len+part3_len+1);
    strlcat(buffer, part3, part1_len+part2_len+part3_len+1);
```

This better?

Reply:
I came across `strlcat` here https://github.com/rbsec/sslscan/issues/267 where it mentions concerns about portability. Historically OpenDNSSEC was tested and built on many platforms but we currently only build and test on Linux, however you recently patched some other code for portability so I thought this worth raising. I defer to your judgement on this.

Reply:
`strlcat` is already used in another place in ods, namely in `signer/src/parser/confparser.c`. More importantly, an `strlcat` implementation is provided (in `common/strlcat.c`) if it is not present on the system on which it is compiled (see `AC_REPLACE_FUNCS(strlcat)` in `configure.ac`).